src/persistence/GenerationPersistence.ts

```diff
@@ -0,0 +1,92 @@
+import {
+    IPersistence,
+    IPersistenceRead,
+} from "@rocket.chat/apps-engine/definition/accessors";
+import {
+    RocketChatAssociationModel,
+    RocketChatAssociationRecord,
+} from "@rocket.chat/apps-engine/definition/metadata";
+
+interface GenerationRecord {
+    query: string;
+    url: string;
+}
+
+interface GenerationRecordWrapper {
+    generated_gifs: GenerationRecord[];
+}
+
+export class GenerationPersistence {
+    private key = "gen-gif";
+
+    constructor(
+        readonly userId: string,
+        readonly persistence: IPersistence,
+        readonly persistenceRead: IPersistenceRead
+    ) {}
+
+    async getAll() {
+        const res = await this.persistenceRead.readByAssociations([
+            new RocketChatAssociationRecord(
+                RocketChatAssociationModel.MISC,
+                this.key
+            ),
+            new RocketChatAssociationRecord(
+                RocketChatAssociationModel.USER,
+                this.userId
+            ),
+        ]);
```

Comment:
@AmanNegi this associations is repeating in entire file, can you make this association to initialise when we create an instance of this ? under constructor make the member of classs as private.